subsys/event_manager/event_manager_shell.c

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <shell/shell.h>
+#include <event_manager/event_manager.h>
+
+static int show_events(const struct shell *event_manager_shell, size_t argc,
+		char **argv)
+{
+	shell_fprintf(event_manager_shell, SHELL_NORMAL,
+		      "Registered Events:\n");
+
+	for (const struct event_type *et = __start_event_types;
+	     (et != NULL) && (et != __stop_event_types); et++) {
+
+		size_t ev_id = et - __start_event_types;
```

Comment:
we should likely avoid using such variables directly and provide iterator functions instead, see for example `device_required_foreach`.